pkg/reconciler/v1alpha1/pipelinerun/pipelinerun.go

```diff
@@ -144,21 +150,74 @@ func (c *Reconciler) Reconcile(ctx context.Context, key string) error {
 
 func (c *Reconciler) reconcile(ctx context.Context, pr *v1alpha1.PipelineRun) error {
 	// fetch the equivelant pipeline for this pipelinerun Run
-	name := pr.Spec.PipelineRef.Name
-	if _, err := c.pipelineLister.Pipelines(pr.Namespace).Get(name); err != nil {
-		c.Logger.Errorf("%q failed to Get Pipeline: %q",
-			fmt.Sprintf("%s/%s", pr.Namespace, pr.Name),
-			fmt.Sprintf("%s/%s", pr.Namespace, name))
-		return nil
+	if _, err := c.getPipelineIfExists(pr); err != nil {
+		return err
 	}
-
 	// TODO fetch the taskruns status for this pipeline run.
 
 	// TODO check status of tasks and update status of PipelineRuns
 
 	return nil
 }
 
+func (c *Reconciler) createPipelineRun(ctx context.Context, p *v1alpha1.Pipeline) error {
+	tasks, err := c.getTasks(p)
+	if err != nil {
+		return errors.NewBadRequest(fmt.Sprintf("pipeline %q is not valid due to %v", p.Name, err))
+	}
+	for _, t := range tasks {
+		tr, err := createTaskRun(t)
+		if err != nil {
+			return err
+		}
+		c.PipelineClientSet.PipelineV1alpha1().TaskRuns(p.Namespace).Create(tr)
+	}
+	return nil
+}
+
+func (c *Reconciler) getTasks(p *v1alpha1.Pipeline) ([]*v1alpha1.Task, error) {
+	tasks := make([]*v1alpha1.Task, len(p.Spec.Tasks))
+	for i, t := range p.Spec.Tasks {
+		tObj, err := c.taskLister.Tasks(p.Namespace).Get(t.TaskRef.Name)
+		if err != nil {
+			c.Logger.Errorf("failed to get tasks for Pipeline %q: Error getting task %q : %s",
+				fmt.Sprintf("%s/%s", p.Namespace, p.Name),
+				fmt.Sprintf("%s/%s", p.Namespace, t.Name), err)
+			return nil, err
+		}
+		tasks[i] = tObj
+	}
+	return tasks, nil
+}
+
+func createTaskRun(t *v1alpha1.Task) (*v1alpha1.TaskRun, error) {
+	// Create empty tasks
+	trName := randomizeTaskRunName(t)
+	tr := &v1alpha1.TaskRun{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      trName,
+			Namespace: t.Namespace,
+		},
+	}
+	return tr, nil
+}
+
+func randomizeTaskRunName(t *v1alpha1.Task) string {
+	return util.AppendRandomString(t.Name)
```

Comment:
can we just use the name of the pipeline in combination with the name of the task?

Reply:
i can add pipeline-run name in combination of name of task hoping we would a unique name for each pipeline-run.